internal/resources/resource_host.go

```diff
@@ -597,13 +597,11 @@ func resourceMetalHostRead(d *schema.ResourceData, meta interface{}) (err error)
 		return err
 	}
 
-	if protocol == "iscsi" {
-		_ = d.Set(hCHAPUser, host.ISCSIConfig.CHAPUser)
-		_ = d.Set(hCHAPSecret, host.ISCSIConfig.CHAPSecret)
-		_ = d.Set(hInitiatorName, host.ISCSIConfig.InitiatorName)
-	}
+	d.Set(hCHAPUser, host.ISCSIConfig.CHAPUser)
+	d.Set(hCHAPSecret, host.ISCSIConfig.CHAPSecret)
+	d.Set(hInitiatorName, host.ISCSIConfig.InitiatorName)
```

Comment:
Looks like iscsi information will always be filled in host here regardless protocol. Is it correct that the protocol field was introduced to differentiate the cases, otherwise it will defeat the purpose.